libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorWebViewAbstract.java

```diff
@@ -43,6 +58,64 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
             public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
                 AppLog.e(AppLog.T.EDITOR, description);
             }
+
+            @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+            @Override
+            public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceRequest request) {
+                String url = request.getUrl().toString();
+
+                try {
+                    // Keep any existing request headers from the WebResourceRequest
+                    Map<String, String> headerMap = request.getRequestHeaders();
+                    for (Map.Entry<String, String> entry : mHeaderMap.entrySet()) {
+                        headerMap.put(entry.getKey(), entry.getValue());
+                    }
+
+                    // Request and add an authorization header for HTTPS resource requests.
+                    // Use https:// when requesting the auth header, in case the resource is incorrectly using http://.
+                    // If an auth header is returned, force https:// for the actual HTTP request.
+                    String authHeader = mAuthHeaderRequestListener.onAuthHeaderRequested(UrlUtils.makeHttps(url));
+                    if (StringUtils.notNullStr(authHeader).length() > 0) {
+                        url = UrlUtils.makeHttps(url);
+                        headerMap.put("Authorization", authHeader);
+                    }
+
+                    HttpURLConnection conn = HTTPUtils.setupUrlConnection(url, headerMap);
```

Comment:
This is not working when the url is "local" - when you try to add a local file for instance, we'll get a `FileURLConnection` and we don't want to call `setupUrlConnection` on it.

So we should check either check for the URL type (if the scheme is `http` or `https`), or  we should check the type (using `instanceOf`) returned by `new URL(url).openConnection()` in `HTTPUtils.setupUrlConnection()`